## Assignments

Now that we have introduced some tools for describing populations, let's try them out. First do these drills by hand, then use the Python code we've provided in the previous assignments to check your work. Keep track of your work in a Google document or markdown file that you can submit below and share with your mentor.

 1. Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch.  Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?
 2. Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?
 3. Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?
 4. Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again.  Does this change your choice of central tendency or variance estimation methods?
 5. On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show.  The answers were:
    TV Guide            20% fans
    Entertainment Weekly    23% fans
    Pop Culture Today       17% fans
    SciPhi Phanatic     5% fans

 Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?

Discuss your answer to each of these questions, along with your code, with your mentor.

When you've given it a try, you can find a solution [here](https://github.com/Thinkful-Ed/data-201-resources/blob/master/solutions/Prep%20course/3.1.4.ipynb).

In [5]:
import numpy as np
import pandas as pd
import statistics
%matplotlib inline


age=[14, 12, 11, 10, 8, 6, 8]
name = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']

#create a black dataframe 
data=pd.DataFrame()

#put data into the dataframe
data['name']=name
data['age']=age

#find mean, median and mode
print("\nMean is {:0.2f}.".format(data['age'].mean()))
print("\nMedian is {:0.2f}.".format(data['age'].median()))
print("\nMode is {:0.2f}.\n".format(statistics.mode(data['age'])))
print(data.describe())
print("\nVariance is {:0.2f}.".format(data.age.var(ddof=0)))
print("\nStandard Deviation is {:0.2f}.".format(data.age.std(ddof=0)))
print(np.std(age))
print("\nStandard Error is {:0.2f}.".format(data.age.sem(ddof=0)))
print(np.std(data['age']) / np.sqrt(len(data['age'])))


Mean is 9.86.

Median is 10.00.

Mode is 8.00.

             age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

Variance is 6.41.

Standard Deviation is 2.53.
2.531435020952764

Standard Error is 0.96.
0.9567925036515135


## Part 1 Mean, Median, Mode, Var, Std, and SE
Mean, median, and mode are 9.86, 10, and 8, respectively.  Variance, Standard deviation, and standard error are 6.41, 2.53, and 0.96, respectively.  Numpy's std function have a default setting of ddof=0, instead of 1 - as it is better used for calculate population std.  Pandas std default is for ddof = 1

## Part 2 Representation of Central Tendency

I'd pick mean and std to represent the central tendency and the variance of the data.  Actually the data is normally distributed and does not have outliners.  Therefore, both mean and median work for central tendency purposes.  

In [6]:
## Cindy is now 7 yo
age[5]=7
data['age']=age

print("\nMean is {:0.2f}.".format(data['age'].mean()))
print("\nMedian is {:0.2f}.".format(data['age'].median()))
print("\nMode is {:0.2f}.\n".format(statistics.mode(data['age'])))
print(data.describe())
print("\nVariance is {:0.2f}.".format(data.age.var(ddof=0)))
print("\nStandard Deviation is {:0.2f}.".format(data.age.std(ddof=0)))
print("\nStandard Error is {:0.2f}.".format(data.age.sem(ddof=0)))



Mean is 10.00.

Median is 10.00.

Mode is 8.00.

             age
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

Variance is 5.43.

Standard Deviation is 2.33.

Standard Error is 0.88.


## Part 3 Cindy's Birthday
After Cindy's birthday, the mean, median and mode are 10, 10, and 8.  Variance, Std, and SE are 5.43, 2.33, and 0.88, respectively.  The median and mode remain the same but all other variables changes. 

In [7]:
age[6]=1
name[6]='Jessica'

data['name']=name
data['age']=age

print("\nMean is {:0.2f}.".format(data['age'].mean()))
print("\nMedian is {:0.2f}.".format(data['age'].median()))
#print("\nMode is {:0.2f}.\n".format(statistics.mode(data['age'])))

try: 
    print("\nMode is {:0.2f}.\n".format(statistics.mode(data['age'])))
except:
    print("\nno mode is found")

print(data.describe())
print("\nVariance is {:0.2f}.".format(data.age.var(ddof=0)))
print("\nStandard Deviation is {:0.2f}.".format(data.age.std(ddof=0)))
print("\nStandard Error is {:0.2f}.".format(data.age.sem(ddof=0)))


Mean is 9.00.

Median is 10.00.

no mode is found
             age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000

Variance is 15.43.

Standard Deviation is 3.93.

Standard Error is 1.48.


## Part 4 Jessica replaces Oliver

Mean and median is 9 and 10.  Mode does not exist.  Variance, Std, and SE are 15.43, 3.93, and 1.48.  I'd pick median to describe the central tendency and use first and 3rd qudratile to describe the variance of the data.  Jessica's age (1) is an extreme outlier in the sample that changes the mean and std drastically.  


## Part 5 Fan
I'd say approximately 20% of the general population are the fans of Brady Bunch.  This is from the general consensus and the data point from Scifi can be considered as an outlier.    